src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -796,6 +796,7 @@ func handleStreamReads(app *app, stream net.Stream, idx int) {
 			bytesToRead := length
 			for bytesToRead > 0 {
 				bufferReadSize := min(MESSAGE_BUFFER_SIZE, bytesToRead)
+				// why ReadFull when we can just use Read?
```

Comment:
`ReadFull` will continously read available data until you have read enough bytes to fill the buffer in the 2nd argument. It essentially calls `Read` in a loop until the buffer is filled, and returns an error if any of the  `Read` calls fail.

Reply:
Yeah, I understand that. Though now we have a loop inside a loop which calls Read. We can simplify this construction to a single loop.

Also I wonder if it is really necessary to send data in chunks to the OCaml process? What about reading the message in full and sending the whole message to the Ocaml process?

I imagine there is nothing too smart on Ocaml side that would justify chunk-based exchange. Correct me if I am wrong, but I expect to find there nothing more than a buffer which concatenates all of the chunks. If so, I really see no difference of whether we do it on Ocaml process side or on Helper side. Removing chunk logic would greatly simplify the protocol and do no harm (unless we do some partial decoding, which I doubt is true), also it will make no difference at all for messages less than 10-100 mb.

Reply:
OK, I think you also propose to get rid of chunking in your IPC refactor RFC :)